### ISSUE TITLE
Potential bug in custom torch cuda ops for alibi-sqrt

### ISSUE THREAD

**Issue by jerryli27** [stepfun-ai/Step-Audio#138]
Dear Step-Audio authors,

Thank you all for the great work and for open sourcing the model. I was trying to make the model work on VLLM, and found that I cannot get the model outputs to match the Step-Audio repo. The logits and the hidden layers seem to drift further away from the original repo's when the input sequence length gets longer. After looking into it a bit more, I suspect that it could be due to the custom CUDA op for the alibi-sqrt calculation. Everything else before the attention layer matches.

I have made [a unit test](https://github.com/stepfun-ai/vllm/commit/08b39e4ddeb0ad5cd0aeb121e7665de8fdf8bcfb) to test out the hypothesis and it seemed that it is indeed the case. The unit test fails most noticeably when alibi_sqrt = True and head_size = 256. There are some other failure cases that I won't go into in detail here. When alibi_sqrt = False and flash attention version = 2, all tests pass, so I believe my testing code should be working fine.

Would you mind looking into the CUDA op you provided, since I do not have your code that produced those ops? Many thanks in advance.

Best,
Jerry

========
机翻：

感谢你们所做的出色工作，以及将模型开源。我一直在尝试让该模型在 VLLM 上运行，结果发现我无法使模型的输出与 Step-Audio 代码库中的结果一致。当输入序列长度变长时，模型的对数几率（logits）和隐藏层与原始代码库中的结果偏差似乎越来越大。在进一步研究之后，我怀疑这可能是由于用于计算 alibi 平方根（alibi-sqrt）的自定义 CUDA 操作造成的。在注意力层之前的其他部分的计算结果都是匹配的。
我做了一个[单元测试](https://github.com/stepfun-ai/vllm/commit/08b39e4ddeb0ad5cd0aeb121e7665de8fdf8bcfb)来验证我的猜想。当 alibi_sqrt = True 且 head_size = 256 时，单元测试的失败情况最为明显。还有一些其他的失败情况，在这里我就不详细阐述了。当 alibi_sqrt = False 且 flash attention 版本为 2 时，所有测试都能通过，所以我认为我的测试代码应该是没问题的。
由于我没有你们生成这些 CUDA 操作的代码，你们介意检查一下你们提供的 CUDA Ops吗？提前万分感谢。

**Comment by Oliver-ss** [comment]
This is our standard pytorch implementation of [alibi-sqrt attention](https://huggingface.co/stepfun-ai/Step-Audio-Chat/blob/main/modeling_step1.py#L39). 
Make sure for attention with kv cache, the attention mask is exectly the same as attention without kv cache. As to your unit test, the relative distance should be `query_len - kv_len - 1`
